pkg/collector/corechecks/cluster/orchestrator/orchestrator_transformer.go

```diff
@@ -91,8 +91,12 @@ func extractService(s *corev1.Service) *model.Service {
 		Status: &model.ServiceStatus{},
 	}
 
-	if s.Spec.IPFamily != nil {
-		message.Spec.IpFamily = string(*s.Spec.IPFamily)
+	if s.Spec.IPFamilies != nil {
+		strFamilies := make([]string, len(s.Spec.IPFamilies))
+		for i, fam := range s.Spec.IPFamilies {
+			strFamilies[i] = string(fam)
+		}
+		message.Spec.IpFamily = strings.Join(strFamilies, ", ")
```

Comment:
In the new vendored version of `kubernetes`, the
```go
        // IPFamily is tombstoned to show why 15 is a reserved protobuf tag.
        // IPFamily *IPFamily `json:"ipFamily,omitempty" protobuf:"bytes,15,opt,name=ipFamily,Configcasttype=IPFamily"`
```
field has been replaced by:
```go
        IPFamilies []IPFamily `json:"ipFamilies,omitempty" protobuf:"bytes,19,opt,name=ipFamilies,casttype=IPFamily"`
```

I didn’t dare reflecting this structure change (`slice` instead of `pointer`) in `model.Service` as it might have an impact on the backend side.

Is concatenating the values like this OK ?

Reply:
concatenating the values should be OK on our side 👍 

@L3n41c do you know how the backward compatibility is handled with previous kube versions?
Should we still check for the old field if the new field is empty?

Reply:
The old field is just not available anymore: https://github.com/kubernetes/api/blob/v0.20.5/core/v1/types.go#L4210